plutus-ledger-api/src/PlutusLedgerApi/V3/Contexts.hs

```diff
@@ -337,24 +337,44 @@ instance PlutusTx.Eq ScriptPurpose where
     a PlutusTx.== a'
   _ == _ = Haskell.False
 
+newtype Lovelace = Lovelace Haskell.Integer
```

Comment:
IMO we should put this in `Value.hs` and use it where we use `Integer` there. There's no representation change since it's a newtype.

Reply:
Placing `Lovelace` next to `Value` type certainly makes sense. However, I don't think there is currently a single function in the `Value.hs` module that could use `Lovelace` as a type. It definitely cannot be used instead of `Integer` in the `Value` type itself, since "Lovelace" is a name for 1/10^6 of ADA, not any other token.

That being said a few helper functions could be useful. Something like:

```haskell
promoteLovelace :: Lovelace -> Value

extractLovelace :: Value -> Maybe Lovelace
```
There is some mention of `'Ledger.Ada.adaValueOf'` in the haddock, but I could not find what exactly it refers to.

Reply:
Yes, I'm being silly, you can't use it in many places in `Value` after all. I did think we had some such helper functions, but perhaps we now don't :thinking:

Reply:
> That being said a few helper functions could be useful.

Does `valueOf v adaSymbol adaToken` return Ada or Lovelace? I assume it's Lovelace but can either of you confirm?

Reply:
Yes, it's lovelace. It always returns the underlying quantity used, which is lovelace. Ada as a unit is "virtual", the quantity tracked in the ledger is always lovelace.

Reply:
Will do separately